web/unit_tests/cases/engine.js

```diff
@@ -38,6 +38,20 @@ function runEngineRuleSet(ruleSet, defaultNoun) {
   }
 }
 
+var toSupplementaryPairString = function(code){
```

Comment:
Could you use source/kmwstring.ts for handling SMP?

Reply:
I don't think these belong in kmwstring.ts, which is specifically setup as extensions to the `string` type (whether or not extending a built-in type is sensible is a separate discussion). They seem fine here.